backend/test/integrated/teamspace.js

```diff
@@ -366,4 +373,304 @@ describe("Checking Quota Info  ", function() {
+				.expect(200, function(err, res) {
+					expect(res.body._id).to.equal(user.user);
+					expect(res.body.riskCategories).to.deep.equal(newRiskCategories);
+					expect(res.body.topicTypes).to.deep.equal(newTopicTypes);
+					done(err);
+				});
+		});
+
+		it("with unexpected field should succeed", function(done) {
+			agent.put(`/${user.user}/settings`)
+				.send({
+					topicTypes: defaultTopicTypeLabels,
+					riskCategories: defaultRiskCategoryLabels,
+					unexpectedField: "abc"
+				})
+				.expect(200, function(err, res) {
+					expect(res.body._id).to.equal(user.user);
+					expect(res.body.riskCategories).to.deep.equal(defaultRiskCategories);
+					expect(res.body.topicTypes).to.deep.equal(defaultTopicTypes);
+					expect(res.body.unexpectedField).to.equal(undefined);
+					done(err);
+				});
+		});
+
+		it("with duplicate risk categories should fail", function(done) {
+			const duplicateRiskCategoryLabels = defaultRiskCategoryLabels.concat(defaultRiskCategoryLabels);
+			agent.put(`/${user.user}/settings`)
+				.send({
+					riskCategories: duplicateRiskCategoryLabels
+				})
+				.expect(400, function(err, res) {
+					expect(res.body.value).to.equal(responseCodes.RISK_DUPLICATE_CATEGORY.value);
+					done(err);
+				});
+		});
+
+		it("with duplicate topic type should fail", function(done) {
+			const duplicateTopicTypeLabels = defaultTopicTypeLabels.concat(defaultTopicTypeLabels);
+			agent.put(`/${user.user}/settings`)
+				.send({
+					topicTypes: duplicateTopicTypeLabels
+				})
+				.expect(400, function(err, res) {
+					expect(res.body.value).to.equal(responseCodes.ISSUE_DUPLICATE_TOPIC_TYPE.value);
+					done(err);
+				});
+		});
+
+		it("with duplicate (case insensitive) categories should fail", function(done) {
+			const duplicateRiskCategoryLabels = ["dup 1", "DUP 1"];
+			agent.put(`/${user.user}/settings`)
+				.send({
+					riskCategories: duplicateRiskCategoryLabels
+				})
+				.expect(400, function(err, res) {
+					expect(res.body.value).to.equal(responseCodes.RISK_DUPLICATE_CATEGORY.value);
+					done(err);
+				});
+		});
+
+		it("with duplicate (case insensitive) topic type should fail", function(done) {
+			const duplicateTopicTypeLabels = ["clone 2", "CLONE 2"];
+			agent.put(`/${user.user}/settings`)
+				.send({
+					topicTypes: duplicateTopicTypeLabels
+				})
+				.expect(400, function(err, res) {
+					expect(res.body.value).to.equal(responseCodes.ISSUE_DUPLICATE_TOPIC_TYPE.value);
+					done(err);
+				});
+		});
+
+		after(function(done) {
+			this.timeout(timeout);
+			agent.post("/logout")
+				.expect(200, done);
+		});
+	});
+
+	describe("Get teamspace settings", function(done) {
+		const user =  imsharedTeamspace;
+
+		before(function(done) {
+			this.timeout(timeout);
+			agent.post("/login")
+				.send({username: user.user, password: user.password})
+				.expect(200, done);
+
+		});
+
+		it("should succeed", function(done) {
+			agent.get(`/${user.user}/settings`)
```

Comment:
Again:
- if username doesn't exist
- If user is not part of teamspace
- if user is not admin